File: Person/react-client/src/dtos/Person.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";

export interface PersonRes {
  personId?: number;
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
  gender?: string;
}

export class PersonReqEdit {
  firstName?: string = "";
  lastName?: string = "";
  phoneNumber?: string = "";
  gender?: string = "";
}

export class PersonReqSearch extends PagedReq {
  gender?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    { gender = "" }
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.gender = gender;
  }
}

File: Person/Person/Dtos/Person.cs
using Person.Common.Paging;
using System.ComponentModel.DataAnnotations;

namespace Person.Dtos
{
    public class PersonRes
    {
        public int PersonId { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string PhoneNumber { get; set; } = "";
        public string Gender { get; set; } = "";
    }

    public class PersonReqEdit 
    {
        [Required, MaxLength(100)]
        public string? FirstName { get; set; }
        [Required, MaxLength(100)]
        public string? LastName { get; set; }
        [MaxLength(20)]
        public string PhoneNumber { get; set; } = "";
        [MaxLength(1)]
        public string Gender { get; set; } = "";
    }
    public class PersonReqSearch : PagedReq 
    {
        [MaxLength(1)]
        public string? Gender { get; set; }
    }
}
